hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/LeaseManager.java

```diff
@@ -626,7 +626,8 @@ private synchronized boolean checkLeases(Collection<Lease> leasesToCheck) {
             }
           }
           // If a lease recovery happened, we need to sync later.
```

Comment:
@vinayakumarb Thank you for your review. Let me explain the current logic. The logic I am modifying now is as follows: if the lease is recovered or the lease is reassigned, it will return false, just like the previous logic. Then, in the checkLeases method, if the return is false and needSync is false, needSync will be reset to true. Subsequently, the edits log will be flushed by leaseMonitor. This way, when RPCs such as recoverLease call the internalReleaseLease method, they can remain consistent with the original behavior.

Reply:
> I would recommend you to change the return type of internalReleaseLease() to ImmutablePair<Boolean, Boolean> to include both completed and needSync values. 
needSync will be true in both cases of file closed and lease re-assignment.

+1. If we will plan to improve it, should fix it together.
BTW, what will it happen if not sync in time, LeaseManager.Monitor is one asynchronous logic, it can not be ensure to sync edits in one certain order right?

Reply:
@Hexiaoqiao @vinayakumarb Thank you very much for your comment. In the one hand,  we may indeed not need to invoke logSync() in time. The purpose of this modification is to ensure that `editlog` can be `sync` in a timely manner like other write operations，so as to prevent the loss of the `editlog` in some extreme cases. on the other hand,  @vinayakumarb I'm still a little confused by what you're saying. The current modification  
```
boolean isClosed = !lastINode.isUnderConstruction();
if (!needSync && (!completed || isClosed)) {
            needSync = true;
          } 
```
 has ensured that leaseMonitor can invoke `logSync()` when the file gets closed and  `reassign lease`. File gets closed, `isClosed` will be true. Lease reassigned , `completed` will be false and the initial value of `needSync` is false.

Reply:
@vinayakumarb  Thank you for your reply. I understand what you mean. But there is another case where calling `logSync()` is not required as mentioned in [HDFS-17519](https://issues.apache.org/jira/browse/HDFS-17519.) Do we need to consider this scenario described in [HDFS-17519](https://issues.apache.org/jira/browse/HDFS-17519)?

Reply:
@vinayakumarb Thank you for your reply. How about changing the method `checkLeases` to return true?